plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -65,606 +68,1719 @@ describe('PersonState.update()', () => {
+                    properties: { b: 4, c: 4, e: 4 },
+                    created_at: timestamp,
+                    version: 1,
+                    is_identified: true,
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['old-user', 'new-user']))
+
+            // verify ClickHouse persons
+            const clickhousePersons = await delayUntilEventIngested(fetchPersonsRows)
+            expect(clickhousePersons.length).toEqual(1)
+            expect(clickhousePersons[0]).toEqual(
+                expect.objectContaining({
+                    id: uuid.toString(),
+                    properties: JSON.stringify({ b: 4, c: 4, e: 4 }),
+                    created_at: timestampch,
+                    version: 1,
+                    is_identified: 1,
+                })
+            )
+
+            // verify ClickHouse distinct_ids
+            await delayUntilEventIngested(fetchDistinctIdsRows)
+            const clickHouseDistinctIds = await hub.db.fetchDistinctIdValues(persons[0], Database.ClickHouse)
+            expect(clickHouseDistinctIds).toEqual(expect.arrayContaining(['old-user', 'new-user']))
+
+            // verify personContainer
+            expect(persons[0]).toEqual(await personContainer.get())
+        })
+        it('merges people when neither identified', async () => {
+            await hub.db.createPerson(timestamp, {}, {}, {}, 2, null, false, uuid.toString(), ['old-user'])
+            await hub.db.createPerson(timestamp2, {}, {}, {}, 2, null, false, uuid2.toString(), ['new-user'])
+
+            const personContainer = await personState({
+                event: '$identify',
+                distinct_id: 'new-user', // note we merge into this person and that's important
+                properties: {
+                    $anon_distinct_id: 'old-user',
+                },
+            }).update()
+            await hub.db.kafkaProducer.flush()
+
+            // verify Postgres persons
+            const persons = await hub.db.fetchPersons()
+            expect(persons.length).toEqual(1)
+            expect(persons[0]).toEqual(
+                expect.objectContaining({
+                    id: expect.any(Number),
+                    uuid: uuid2.toString(),
+                    properties: {},
+                    created_at: timestamp,
+                    version: 1,
+                    is_identified: true,
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['old-user', 'new-user']))
+
+            // verify ClickHouse persons
+            const clickhousePersons = (await delayUntilEventIngested(() => fetchPersonsRows(), 2)).sort(
```

Comment:
Needing to do the sort here just results in noise in the code, making everything harder to follow. Coupled with this test being 60 (!) lines it's really hard to tell what's important here and what isn't.

Let's use primitives that don't add noise like `expect.arrayContaining` or make the `fetchPersonRows` sort by something!

Reply:
I'll move that into the sort up - will order by _offset in the fetch ch person rows function.
postgres one: helper in this class